src/ci.js

```diff
@@ -1,30 +1,56 @@
 const { v4: uuidv4 } = require('uuid')
-const { name, version } = require('.././package.json')
+const { name: collector, version } = require('.././package.json')
 
 class CI {
+  // the analytics env are more specific than the automatic ci platform env.
+  // If they've been specified we'll assume the user wants to use that value instead.
   env() {
     return({
-      "version": version,
-      "collector": `js-${name}`,      
-      ...this.ci_env()
+      ...this.ci_env(),
+      ...this.analytics_env()
+    })
+  }
+
+  _stripUndefinedKeys(object) {
+    Object.keys(object).forEach((key) => {
+      if (object[key] === undefined) {
+        delete object[key]
+      }
     })
+
+    return object
   }
 
   ci_env() {
-    if (process.env.BUILDKITE_BUILD_ID !== undefined) {
-      return(this.buildkite())
-    } else if (process.env.GITHUB_RUN_NUMBER !== undefined) {
-      return(this.github_actions())
-    } else if (process.env.CIRCLE_BUILD_NUM  !== undefined) {
-      return(this.circleci())
-    } else {
-      return(this.generic())
-    }
+    if(process.env.BUILDKITE_BUILD_ID !== undefined) { return(this.buildkite()) }
+    if(process.env.GITHUB_RUN_NUMBER !== undefined) { return(this.github_actions()) }
+    if(process.env.CIRCLE_BUILD_NUM  !== undefined) { return(this.circleci()) }
+    if(process.env.CI !== undefined) { return(this.generic()) }
+
+    return({
+      'CI': undefined,
+      'key': uuidv4()
+    })
+  }
+
+  analytics_env() {
+    return this._stripUndefinedKeys({
+      "key": process.env.BUILDKITE_ANALYTICS_KEY,
```

Comment:
Sorry to throw my two cents in...

From the [docs](https://buildkite.com/docs/test-analytics/integrations#importing-json) the key should be the build id.

```javascript
      "key": process.env.BUILDKITE_BUILD_ID,
```
note that this would override the key set in `buildkte()` below.

Reply:
Hey @HookyQR! I was checking out your Jest collector, be great to have you contributing to this one.

Good spotting on this bug being reintroduced, that was the original intention for the changes in #1.

Reply:
(your two cents is very much appreciated 😊)

Reply:
Thanks @HookyQR this value is intended to be an override of the value inferred from the CI platform. If someone is wishing to use the `BUILDKITE_BUILD_ID` inferred from the Buildkite env they can leave `BUILDKITE_ANALYTICS_KEY` blank.

This ensures the flexibility of customization and the simplicity of turn-key operation on Buildkite and other CI platforms.